File: JavaToPython/QueryWriterTests.py
from .SimpleQueryWriter import SimpleQueryWriter


def test_stopword_removal(writer: SimpleQueryWriter):
    q = "Bu ders nedir ve nasıl alınır?"
    terms = writer.write(q, None)
    stops = ["bu", "nedir", "nasıl", "ve"]
    passed = all(s not in terms for s in stops)
    print(("PASS" if passed else "FAIL") + f": Stopword removal -> {terms}")


def test_intent_boosters(writer: SimpleQueryWriter):
    q = "Bilgi"
    course_terms = writer.write(q, None)
    has_booster = "ders" in course_terms or "kredi" in course_terms
    print(("PASS" if has_booster else "FAIL") + f": Course -> {course_terms}")


def test_stemming_behavior(writer: SimpleQueryWriter):
    q = "Öğretmenleri bulmak istiyorum"
    terms = writer.write(q, None)
    contains_inflected = any("öğretmen" in t for t in terms)
    contains_booster_base = "öğretmen" in terms or "öğretim" in terms
    passed = contains_inflected and contains_booster_base
    print(("PASS" if passed else "FAIL") + f": terms={terms}")


if __name__ == "__main__":
    writer = SimpleQueryWriter()
    test_stopword_removal(writer)
    test_intent_boosters(writer)
    test_stemming_behavior(writer)

File: Java/QueryWriterTests.java
package mu.cse.rag.clean;

import java.util.List;

public class QueryWriterTests {
    public static void main(String[] args) {
        SimpleQueryWriter writer = new SimpleQueryWriter();

        testStopwordRemoval(writer);
        testIntentBoosters(writer);
        testStemmingBehavior(writer);
    }

    private static void testStopwordRemoval(SimpleQueryWriter writer) {
        String q = "Bu ders nedir ve nasıl alınır?";
        List<String> terms = writer.write(q, Intent.COURSE);
        boolean pass = true;
        String[] stops = {"bu", "nedir", "nasıl", "ve"};
        for (String s : stops) {
            if (terms.contains(s)) {
                pass = false;
            }
        }
        System.out.println((pass ? "PASS" : "FAIL") + ": Stopword removal -> " + terms);
    }

    private static void testIntentBoosters(SimpleQueryWriter writer) {
        String q = "Bilgi";
        List<String> courseTerms = writer.write(q, Intent.COURSE);
        boolean hasBooster = courseTerms.contains("ders") || courseTerms.contains("kredi");
        System.out.println((hasBooster ? "PASS" : "FAIL") + ": Course -> " + courseTerms);

        List<String> staffTerms = writer.write(q, Intent.STAFF_LOOKUP);
        boolean hasStaff = staffTerms.contains("öğretim") || staffTerms.contains("hoca");
        System.out.println((hasStaff ? "PASS" : "FAIL") + ": Staff -> " + staffTerms);
    }

    private static void testStemmingBehavior(SimpleQueryWriter writer) {
        String q = "Öğretmenleri bulmak istiyorum";
        List<String> terms = writer.write(q, Intent.STAFF_LOOKUP);
        boolean containsInflected = terms.stream().anyMatch(t -> t.contains("öğretmen"));
        boolean containsBoosterBase = terms.contains("öğretmen") || terms.contains("öğretim");
        boolean pass = containsInflected && containsBoosterBase;
        System.out.println((pass ? "PASS" : "FAIL") + ": terms=" + terms);
    }
}
